ibis/backends/risingwave/__init__.py

```diff
@@ -0,0 +1,281 @@
+        driver
+            Database driver
+
+        Examples
+        --------
+        >>> import os
+        >>> import getpass
+        >>> import ibis
+        >>> host = os.environ.get("IBIS_TEST_RISINGWAVE_HOST", "localhost")
+        >>> user = os.environ.get("IBIS_TEST_RISINGWAVE_USER", getpass.getuser())
+        >>> password = os.environ.get("IBIS_TEST_RISINGWAVE_PASSWORD")
+        >>> database = os.environ.get("IBIS_TEST_RISINGWAVE_DATABASE", "dev")
+        >>> con = connect(database=database, host=host, user=user, password=password)
+        >>> con.list_tables()  # doctest: +ELLIPSIS
+        [...]
+        >>> t = con.table("functional_alltypes")
+        >>> t
+        RisingwaveTable[table]
+          name: functional_alltypes
+          schema:
+            id : int32
+            bool_col : boolean
+            tinyint_col : int16
+            smallint_col : int16
+            int_col : int32
+            bigint_col : int64
+            float_col : float32
+            double_col : float64
+            date_string_col : string
+            string_col : string
+            timestamp_col : timestamp
+            year : int32
+            month : int32
+        """
+        if driver != "psycopg2":
+            raise NotImplementedError("psycopg2 is currently the only supported driver")
+
+        alchemy_url = self._build_alchemy_url(
+            url=url,
+            host=host,
+            port=port,
+            user=user,
+            password=password,
+            database=database,
+            driver=f"risingwave+{driver}",
+        )
+
+        connect_args = {}
+        if schema is not None:
+            connect_args["options"] = f"-csearch_path={schema}"
+
+        engine = sa.create_engine(
+            alchemy_url, connect_args=connect_args, poolclass=sa.pool.StaticPool
+        )
+
+        @sa.event.listens_for(engine, "connect")
+        def connect(dbapi_connection, connection_record):
+            with dbapi_connection.cursor() as cur:
+                cur.execute("SET TIMEZONE = UTC")
+
+        super().do_connect(engine)
+
+    def list_tables(self, like=None, schema=None):
+        """List the tables in the database.
+
+        Parameters
+        ----------
+        like
+            A pattern to use for listing tables.
+        schema
+            The schema to perform the list against.
+
+            ::: {.callout-warning}
+            ## `schema` refers to database hierarchy
+
+            The `schema` parameter does **not** refer to the column names and
+            types of `table`.
+            :::
+        """
+        tables = self.inspector.get_table_names(schema=schema)
+        views = self.inspector.get_view_names(schema=schema)
+        return self._filter_with_like(tables + views, like)
+
+    def list_databases(self, like=None) -> list[str]:
+        # http://dba.stackexchange.com/a/1304/58517
+        dbs = sa.table(
+            "pg_database",
```

Comment:
I think `pg` here is for `postgres`. Is this intentional? Do we want to change this to `risingwave` or something to avoid confusion?

Reply:
`pg_database` is [a part of pg's catalog](https://www.postgresql.org/docs/current/catalog-pg-database.html). Risingwave support both `rw_catalog` and `pg_catalog`(partially). Here, using `pg_database` is OK.